src/edu/csus/ecs/pc2/core/report/ContestCompareModel.java

```diff
@@ -0,0 +1,821 @@
+                // EOF
+                break;
+            }
+
+            Map<String, Object> eventMap = JSONUtilities.getMap(event);
+            ObjectMapper mapper = getMapper();
+
+            if (eventMap == null) {
+                // could not parse event
+                getLog().log(Level.WARNING, "Could not parse event: " + event);
+                if (Utilities.isDebugMode()) {
+                    System.out.println("Could not parse event: " + event);
+                }
+
+            } else {
+
+                CLICSEventFeedEvent eventFeedEntry = (CLICSEventFeedEvent) mapper.readValue(event, CLICSEventFeedEvent.class);
+                // find event type
+                String eventType = eventFeedEntry.getType();
+
+//                if (isAfterConfig(eventType)) {
+//                    break;
+//                }
+
+//                System.out.println(" event "+eventType+" "+eventFeedEntry.getData());
+
+                if (eventFeedEntry.getData() == null) {
+                    
+                    // SOMEDAY handle delete or whatver op has null data
+                    // log for now.
+                    getLog().log(Level.FINE, "Line: "+eventCount+ " data:null in event line '"+event+"'");
+                    
+                }
+                else if (CLICSEventType.LANGUAGES.toString().equals(eventType)) {
+                    // languages
+
+                    // languages data = {name=C, id=c}
+
+                    CLICSLanguage language = mapper.convertValue(eventFeedEntry.getData(), CLICSLanguage.class);
+
+                    feedLanguages.add(language);
+
+                } else if (CLICSEventType.JUDGEMENT_TYPES.toString().equals(eventType)) {
+                    // judgement-types
+
+                    // {"type": "judgement-types", "id": "a17b7d9d-43fc-43ed-bfd1-ef7fccbc6feb", "op": "create", "data": {"id": "RTE", "name": "Run Time Error", "solved": false, "penalty": true}}
+
+                    JudgementType judgementType = mapper.convertValue(eventFeedEntry.getData(), JudgementType.class);
+                    feedJudgements.add(judgementType);
+
+                } else if (CLICSEventType.PROBLEMS.toString().equals(eventType)) {
+                    // problems
+
+                    // {time_limit=3, color=null, name=Particle Swapping, id=particles, label=F, test_data_count=9, rgb=null, ordinal=6}
+
+                    CLICSProblem problem = mapper.convertValue(eventFeedEntry.getData(), CLICSProblem.class);
+
+                    // System.out.println("type " + eventType + " data = " + eventFeedEntry.getData().toString());
+
+                    feedProblems.add(problem);
+                } else if (CLICSEventType.TEAMS.toString().equals(eventType)) {
+                    // teams
+
+                    // {"type": "teams", "id": "289de609-b1a0-40d8-a9e0-54a30e752e69", "op": "create", "data": {"id": "169347", "name": "Colin Peppler", "icpc_id": null, "group_ids": [],
```

Comment:
This is a bad example to have in the code as it is for the old feed.  Suggest updating the comment to use a 2022-07 team event.

Reply:
It's what I had 4 weeks ago when I started.   I expected to make these sort of updates nearly 2 weeks ago when the PR was created.   But I have not attempted to track/analyse the content because that can be easily updated.  It was work to get the framework and content/layout done.